corehq/apps/app_manager/views/utils.py

```diff
@@ -197,14 +198,20 @@ def overwrite_app(app, master_build, report_map=None):
     wrapped_app = _update_forms(wrapped_app, master_build, ids_map)
 
     # Multimedia versions should be set based on the linked app's versions, not those of the master app.
-    for path in wrapped_app.multimedia_map.keys():
-        wrapped_app.multimedia_map[path].version = None
     wrapped_app.set_media_versions()
 
     enable_usercase_if_necessary(wrapped_app)
     return wrapped_app
 
 
+def _update_multimedia_map(old_map, new_map):
+    for path in new_map:
```

Comment:
I'm not sure if this scenario is possible but we should make sure that none of the items in 'new_map' have 'upstream_media_id' attributes (or at least we should remove them).